src/components/fields/SchemaField.js

```diff
@@ -147,6 +148,10 @@ function SchemaField(props) {
     return <div/>;
   }
 
+  if (theme) {
+    mergeToDefaultWidgets(theme);
```

Comment:
That's a massive side effect. I'd rather at least have a new object retrieved from the call to this function:

```js
const themedWidgets = mergeToDefaultWidgets(theme);
```